### PR TITLE
BME688 support

### PR DESCRIPTION
**Pull request by CHH**
Added the same changes as in the pimoroni repo to support the gas measurements on the BME688. The variant is automatically detected.

I didn't add tests for now, because I'm not familiar with how it should be done. Help would be appreciated.

### CODE REVIEW

**Comment by ftmazzone** [comment]
@CHH Thanks, sure I will be happy to add support for BME688. 

For the unit test, I've added a configuration file to debug the existing unit tests.
To ensure that the coverage remains the same, you can use `npm run coverage`.

Could you add a few comments in the constructor to understand the mapping bme68x <-> CHIP_VARIANT (LOW or HIGH) ?

**Review by ftmazzone — COMMENTED** [review]
Could you increase the `package.version` number to 1.0.2 ?

One new property chip_name ('bme680'/'bme688') can also be added to [bme680Data](https://github.com/ftmazzone/bme680/blob/master/lib/bme680Data.js#L21 ).

**Comment by CHH** [comment]
@ftmazzone I'll look into it in the coming week!

File: lib/constants.js [hunk 1]
```diff
@@ -147,7 +154,7 @@ module.exports = class Constants {
         this.OSP_MSK = 0X1C;
         this.OSH_MSK = 0X07;
         this.HCTRL_MSK = 0x08;
-        this.RUN_GAS_MSK = 0x10;
+        this.RUN_GAS_MSK = 0x30;
```
Comment (ftmazzone) [inline lib/constants.js:157]: is this value different between the BME680 and BME688 ?

File: lib/bme680.js [hunk 2]
```diff
@@ -74,27 +87,42 @@ module.exports = class Bme680 {
...
         const adc_hum = (regs[8] << 8) | regs[9];
-        const adc_gas_res = (regs[13] << 2) | (regs[14] >> 6);
-        const gas_range = regs[14] & constants.GAS_RANGE_MSK;
 
-        bme680Data.data.status |= regs[14] & constants.GASM_VALID_MSK;
-        bme680Data.data.status |= regs[14] & constants.HEAT_STAB_MSK;
+        const adc_gas_res_low = (regs[13] << 2) | (regs[14] >> 6);
+        const adc_gas_res_high = (regs[15] << 2) | (regs[16] >> 6);
+
+        const gas_range_l = regs[14] & constants.GAS_RANGE_MSK;
```
Comment (ftmazzone) [inline lib/bme680.js:94]: can we move the initialisation of the `gas_range` and `adc_gas_res` value inside the next if statement ? 


```
let gas_range,adc_gas_res ;
        if (this.chip_variant === constants.VARIANT_HIGH) {
            bme680Data.data.status |= regs[16] & constants.GASM_VALID_MSK;
            bme680Data.data.status |= regs[16] & constants.HEAT_STAB_MSK;
            gas_range = regs[16] & constants.GAS_RANGE_MSK;
            adc_gas_res = (regs[15] << 2) | (regs[16] >> 6);
        } else {
            bme680Data.data.status |= regs[14] & constants.GASM_VALID_MSK;
            bme680Data.data.status |= regs[14] & constants.HEAT_STAB_MSK;
            gas_range = regs[14] & constants.GAS_RANGE_MSK;
            adc_gas_res = (regs[13] << 2) | (regs[14] >> 6);
        }
```

File: lib/bme680.js [hunk 3]
```diff
@@ -74,27 +87,42 @@ module.exports = class Bme680 {
...
         bme680Data.data.temperature = temperature / 100.0;
-        this.ambient_temperature = temperature;// Saved for heater calc;
+        this.ambient_temperature = temperature; // Saved for heater calc;
         bme680Data.ambient_temperature = this.ambient_temperature;
 
         bme680Data.data.pressure = await this.calcPressure(adc_pres) / 100.0;
         bme680Data.data.humidity = await this.calcHumidity(adc_hum) / 1000.0;
-        bme680Data.data.gas_resistance = await this.calcGasResistance(adc_gas_res, gas_range);
+
+        if (this.chip_variant === constants.VARIANT_HIGH) {
```
Comment (ftmazzone) [inline lib/bme680.js:115]: we can use `calcGasResistance`

File: README.md [hunk 4]
```diff
@@ -11,7 +11,7 @@ Node.js module for controlling [Bosch Sensortec BME680](https://ae-bst.resource.
 ## Installation
 
 ```sh
-npm install bme680-sensor
+npm install github:CHH/bme680
```
Comment (ftmazzone) [inline README.md:14]: needs to be reverted

File: lib/bme680.js [hunk 5]
```diff
@@ -23,22 +23,35 @@ module.exports = class Bme680 {
...
             throw new Error(invalidChipIdError);
         }
+
+        this.chip_variant = await this.readByte(constants.CHIP_VARIANT_ADDR, 1);
+
+        if (this.chip_variant !== constants.VARIANT_LOW && this.chip_variant !== constants.VARIANT_HIGH) {
+            throw new Error(`BME680 Invalid chip variant: ${this.chip_variant}`)
+        }
+
         bme680Data.chip_id = constants.CHIP_ID;
```
Comment (ftmazzone) [inline lib/bme680.js:41]: `bme680Data` cannot be static anymore as a bme680 and bme688 can run in parallel. It needs to be initialised in the constructor.

**Comment by ftmazzone** [comment]
@CHH I've added some unit tests. Could you check if the data reported by the sensor bme688 are correct ? I have only a bme680 and won't be able to test the bme688.

**Review by ftmazzone — APPROVED** [review]

**Pull request merged**